sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/implementation/DeploymentSlotImpl.java

```diff
@@ -213,40 +211,24 @@ public CsmDeploymentStatus getDeploymentStatus(String deploymentId) {
     @Override
     public Mono<CsmDeploymentStatus> getDeploymentStatusAsync(String deploymentId) {
         // "GET" LRO is not supported in azure-core
-        AtomicLong pollCount = new AtomicLong();
-        Duration pollDuration = manager().serviceClient().getDefaultPollInterval();
+        SerializerAdapter serializerAdapter = SerializerFactory.createDefaultManagementSerializerAdapter();
         return this.manager().serviceClient().getWebApps()
             .getProductionSiteDeploymentStatusWithResponseAsync(this.resourceGroupName(), this.name(), deploymentId)
             .flatMap(fluxResponse -> {
-                if (pollDuration.multipliedBy(pollCount.get()).compareTo(MAX_DEPLOYMENT_STATUS_TIMEOUT) < 0) {
-                    HttpResponse response = new HttpFluxBBResponse(fluxResponse);
-                    if (response.getStatusCode() / 100 != 2) {
-                        return Mono.error(new ManagementException("Service responds with a non-20x response.", response));
-                    }
-                    return response.getBodyAsString()
-                        .flatMap(bodyString -> {
-                            SerializerAdapter serializerAdapter = JacksonAdapter.createDefaultSerializerAdapter();
-                            CsmDeploymentStatus status;
-                            try {
-                                status = serializerAdapter.deserialize(bodyString, CsmDeploymentStatus.class, SerializerEncoding.JSON);
-                            } catch (IOException e) {
-                                return Mono.error(new ManagementException("Deserialize failed for response body.", response));
-                            }
-                            if (status == null) {
-                                return Mono.empty();
-                            }
-                            return Mono.just(status);
-                        });
-                } else {
-                    return Mono.error(new ManagementException("Timeout getting deployment status for deploymentId: " + deploymentId, null));
+                HttpResponse response = new HttpFluxBBResponse(fluxResponse);
+                if (fluxResponse.getStatusCode() / 100 != 2) {
+                    return Mono.error(new ManagementException("Service responds with a non-20x response.", response));
                 }
```

Comment:
I think the client API should already handle the error case (actually the underlying proxy method).